pinot-spi/src/test/java/org/apache/pinot/spi/utils/QueryOptionsUtilsTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.spi.utils;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class QueryOptionsUtilsTest {
+
+  @Test
+  public void shouldConvertCaseInsensitiveMapToUseCorrectValues() {
+    // Given:
+    Map<String, String> configs = ImmutableMap.of(
+        "ENABLENullHandling", "true",
+        "useMULTISTAGEEngine", "false"
+    );
```

Comment:
let's also add a test for option key-value pairs that are not in the QueryOptionKey set
1. with underscores
2. with partial match
3. completely irrelevant. 
4. dot separated and some escaped characters. 

the goal is to show that only those in QueryOptionKey are case insensitive. others are not